src/peft/peft_model.py

```diff
@@ -551,10 +551,13 @@ def load_adapter(self, model_id: str, adapter_name: str, is_trainable: bool = Fa
                     )
 
         if use_safetensors:
-            adapters_weights = safe_load_file(filename, device="cuda" if torch.cuda.is_available() else "cpu")
+            adapters_weights = safe_load_file(
+                filename, device=kwargs.get("device_map", "cuda" if torch.cuda.is_available() else "cpu")
```

Comment:
What if someone passes `device_map="auto"`? I am not sure this change is robust enough

Reply:
Note also that `device_map` can be also a dict of strs